grc/gui_qt/Utils.py

```diff
@@ -97,14 +101,18 @@ def make_screenshot(fg_view, file_path, transparent_bg=False):
         except ImportError:
             log.error("Missing (Python-)QtSvg! Please install it or export as PNG instead.")
             return
-        rect = fg_view.viewport().rect()
 
         generator = QtSvg.QSvgGenerator()
         generator.setFileName(str(file_path))
-        generator.setSize(rect.size())
         painter = QtGui.QPainter(generator)
-        fg_view.render(painter)
+        fg_view.render(painter, target_rect, source_rect)
         painter.end()
     elif file_path.suffix == ".pdf":
-        log.warning("PDF screen capture not implemented")
-        return  # TODO
+        pdf_printer = QPrinter(QPrinter.ScreenResolution)
```

Comment:
It looks like there has been some changes from Qt5 to Qt6, I'm getting errors from several of these lines when running with `export QT_API="pyqt6"`. 

I was able to fix it (for both 5 and 6) with the following changes:
```python
        pdf_printer = QPrinter(QPrinter.PrinterMode.ScreenResolution)
        pdf_printer.setOutputFormat(QPrinter.OutputFormat.PdfFormat)
        pdf_printer.setOutputFileName(str(file_path))
        page_layout = QPageLayout()
        page_layout.setOrientation(QPageLayout.Landscape)
        pdf_printer.setPageLayout(page_layout)
        painter = QtGui.QPainter(pdf_printer)
        fg_view.render(painter, target_rect, source_rect)
        painter.end()
        return
```

Reply:
Yes there are differences between 5 and 6 and with your changes the pdf creation works for both.
But the output is different to the earlier version. Now the fg scene will only be partially printed due to some scaling.
Changing the fg_view.render liner did the trick